src/libOpenImageIO-C/c-imageio.cpp

```diff
@@ -0,0 +1,257 @@
+                            value, casesensitive);
+}
+
+
+
+// FIXME: add Filesystem::IOProxy
+ImageInput*
+ImageInput_open(const char* filename, const ImageSpec* config, void* _ioproxy)
+{
+    auto ii = ImageInput::open(filename, config, nullptr);
+    return ii.release();
+}
+
+
+
+bool
+ImageInput_close(ImageInput* ii)
+{
+    return ii->close();
+}
+
+
+
+void
+ImageInput_delete(ImageInput* ii)
+{
+    delete ii;
+}
+
+
+
+const ImageSpec*
+ImageInput_spec(ImageInput* ii)
+{
+    return &ii->spec();
+}
+
+
+
+bool
+ImageInput_read_image(ImageInput* ii, int subimage, int miplevel, int chbegin,
+                      int chend, TypeDesc format, void* data, stride_t xstride,
+                      stride_t ystride, stride_t zstride,
+                      ProgressCallback progress_callback,
+                      void* progress_callback_data)
+{
+    return ii->read_image(subimage, miplevel, chbegin, chend,
+                          bit_cast<TypeDesc, OIIO::TypeDesc>(format), data,
+                          xstride, ystride, zstride, progress_callback,
+                          progress_callback_data);
+}
+
+
+
+bool
+ImageInput_has_error(const ImageInput* ii)
+{
+    return ii->has_error();
+}
+
+
+
+void
+ImageInput_geterror(const ImageInput* ii, char* msg, int buffer_length,
+                    bool clear)
+{
+    std::string errorstring = ii->geterror(clear);
+    int length = std::min(buffer_length, (int)errorstring.size() + 1);
+    memcpy(msg, errorstring.c_str(), length);
+    msg[length - 1] = '\0';
+}
+
+
+
+ImageOutput*
+ImageOutput_create(const char* filename, void* ioproxy,
+                   const char* plugin_search_path)
+{
+    return ImageOutput::create(filename, nullptr, plugin_search_path).release();
+}
+
+
+
+void
+ImageOutput_delete(ImageOutput* io)
+{
+    delete io;
+}
+
+
+
+bool
+ImageOutput_open(ImageOutput* io, const char* name, const ImageSpec* newspec,
+                 int mode)
+{
+    return io->open(std::string(name), *newspec,
+                    (OIIO::ImageOutput::OpenMode)mode);
+}
+
+
+
+bool
+ImageOutput_has_error(const ImageOutput* io)
+{
+    return io->has_error();
+}
+
+
+
+void
+ImageOutput_geterror(const ImageOutput* io, char* msg, int buffer_length,
+                     bool clear)
+{
+    std::string errorstring = io->geterror(clear);
+    int length = std::min(buffer_length, (int)errorstring.size() + 1);
+    memcpy(msg, errorstring.c_str(), length);
+    msg[length - 1] = '\0';
+}
+
+
+
+bool
+ImageOutput_write_image(ImageOutput* io, TypeDesc format, const void* data,
+                        stride_t xstride, stride_t ystride, stride_t zstride,
+                        ProgressCallback progress_callback,
+                        void* progress_callback_data)
+{
+    return io->write_image(bit_cast<TypeDesc, OIIO::TypeDesc>(format), data,
+                           xstride, ystride, zstride, progress_callback,
+                           progress_callback_data);
+}
+int
```

Comment:
blank lines go here

Reply:
Yep sorry about that, I was having a weird issue in vscode yesterday where clang-format was barfing when I had a trailing newline. I've found the appropriate setting to add those automatically and all seems good now.